pkg/internal/candidatebuffer/candidatebuffer.go

```diff
@@ -85,7 +89,9 @@ func (c *CandidateBuffer) BufferStream(ctx context.Context, incoming <-chan type
 			if c.afterEach != nil {
 				c.afterEach <- struct{}{}
```

Comment:
should we put these in a select that allows for context cancellation too so we don't get blocked on this? I don't recall what we're doing with these but it seems like this has potential to hold up returns too

Reply:
just test code, not super important